warehouse/query-core/src/test/java/datawave/query/util/DateIndexHelperTest.java

```diff
@@ -225,4 +304,53 @@ public void testDateIndexHelperHint() throws Exception {
                         DateIndexUtil.getBeginDate("20090101"), DateIndexUtil.getEndDate("20120101"), Collections.singleton("test"));
         Assert.assertEquals("", hint);
     }
+
+    @Test
+    public void testDateIndexHelperHintWithIterator() throws Exception {
+        DateIndexHelper helper = this.dateIndexHelperFactory.createDateIndexHelper().initialize(client, TableName.DATE_INDEX, Collections.singleton(auths), 2,
+                        0.9f);
+        helper.setUseIterator(true);
+
```

Comment:
Can you add some documentation about what scenarios/edge cases each of these different asserts are verifying?